### DISCUSSION TITLE
PR autoclosed and reopened without any reason

### DISCUSSION THREAD

**Discussion by rknj** [renovatebot/renovate#19801] [Request Help] [answered]
I noticed this in several PRs:
![image](https://user-images.githubusercontent.com/17745190/212013757-31625fd5-6b9e-470e-955e-4b69ab6f702a.png)

With these logs in the dashboard, when it was closed:

> DEBUG: findPr(renovate/maven-org.yaml-snakeyaml-vulnerability, undefined, open)
> DEBUG: Found PR #1459
> DEBUG: branch.isModified(): using git to calculate
> DEBUG: branch.isModified() = false
> DEBUG: setCachedModifiedResult(): Branch cache not present
> INFO: Autoclosing PR
> {
>   "branchName": "renovate/maven-org.yaml-snakeyaml-vulnerability",
>   "prNo": 1459,
>   "prTitle": "Update dependency org.yaml:snakeyaml to v1.32 [SECURITY]"
> }
> DEBUG: updatePr(1459, Update dependency org.yaml:snakeyaml to v1.32 [SECURITY] - autoclosed, body)
> DEBUG: PR updated...prNo: 1459
> DEBUG: Deleted remote branch: renovate/maven-org.yaml-snakeyaml-vulnerability

And this when it has been reopened:

> DEBUG: getBranchPr(renovate/maven-org.yaml-snakeyaml-vulnerability)
> DEBUG: findPr(renovate/maven-org.yaml-snakeyaml-vulnerability, undefined, open)
> DEBUG: findPr(renovate/maven-org.yaml-snakeyaml-vulnerability, undefined, closed)
> DEBUG: Found PR #1459
> DEBUG: Found autoclosed PR for branch
> {
>   "autoclosedPr": {
>     "displayNumber": "Pull Request #1459",
>     "number": 1459,
>     "sourceBranch": "renovate/maven-org.yaml-snakeyaml-vulnerability",
>     "title": "Update dependency org.yaml:snakeyaml to v1.32 [SECURITY] - autoclosed",
>     "state": "closed",
>     "bodyStruct": {
>       "hash": "e4980f79fe4436fd8fd00ffcececd8057275be1aee7e460eff725c2769f64ed2",
>       "rebaseRequested": false,
>       "debugData": {
>         "createdInVer": "34.97.1",
>         "updatedInVer": "34.97.1"
>       }
>     },
>     "updated_at": "2023-01-12T04:25:12Z",
>     "node_id": "PR_kwDOA7uynM5HEqVp",
>     "sha": "d51f1efd75f1bc81c892d8d8ee4695a1a62c7514",
>     "sourceRepo": "Jahia/xxxxx",
>     "labels": [
>       "dependencies"
>     ],
>     "hasReviewers": true,
>     "createdAt": "2023-01-10T13:53:15Z",
>     "closedAt": "2023-01-12T04:25:10Z"
>   }
> }
> DEBUG: Recreated autoclosed branch renovate/maven-org.yaml-snakeyaml-vulnerability with sha d51f1efd75f1bc81c892d8d8ee4695a1a62c7514
> INFO: Successfully reopened autoclosed PR
> {
>   "branchName": "renovate/maven-org.yaml-snakeyaml-vulnerability",
>   "title": "Update dependency org.yaml:snakeyaml to v1.32 [SECURITY]",
>   "number": 1459
> }

There were no changes in the base branch since the auto-creation of the PR.
Can someone explain me what is going on?

Here's the config:

> {
>   "$schema": "https://docs.renovatebot.com/renovate-schema.json",
>   "extends": ["config:base"],
>   "labels": ["dependencies"],
>   "packageRules": [
>     {
>       "updateTypes": ["minor", "patch"],
>       "automerge": false
>     },
>     {
>       "matchPackagePatterns": ["^eslint"],
>       "enabled": false
>     },
>     {
>       "matchDepTypes": ["optionalDependencies"],
>       "addLabels": ["optional"]
>     },
>     {
>       "matchDepTypes": ["devDependencies"],
>       "enabled": false
>     }
>   ],
>   "timezone": "Europe/Paris",
>   "schedule": ["every 3 months on the first day of the month"]
> }

**Comment by rarkins** [comment]
Need more logs from those runs to compare

**Reply by rknj** [reply]
Here are additional logs from the run where the PR have been reopened: [renovate_run_969188199_PR_reopen.log](https://github.com/renovatebot/renovate/files/10411605/renovate_run_969188199_PR_reopen.log)

For the run where the PR has been autoclosed, I did not find any other log entries mentionning the library nor the PR number

**Reply by rknj** [reply]
I saw these logs btw (related to `org.yaml-snakeyaml-1.x` and not `maven-org.yaml-snakeyaml-vulnerability`), don't know if that could help: 
> DEBUG: getBranchPr(renovate/org.yaml-snakeyaml-1.x)
> DEBUG: findPr(renovate/org.yaml-snakeyaml-1.x, undefined, open)
> DEBUG: findPr(renovate/org.yaml-snakeyaml-1.x, undefined, closed)
> ...
> DEBUG: syncBranchState()(branch="renovate/org.yaml-snakeyaml-1.x")
> DEBUG: syncBranchState(): Branch cache not found, creating minimal branchState(branch="renovate/org.yaml-snakeyaml-1.x")
> DEBUG: getBranchPr(renovate/org.yaml-snakeyaml-1.x)(branch="renovate/org.yaml-snakeyaml-1.x")
> DEBUG: findPr(renovate/org.yaml-snakeyaml-1.x, undefined, open)(branch="renovate/org.yaml-snakeyaml-1.x")
> DEBUG: findPr(renovate/org.yaml-snakeyaml-1.x, undefined, closed)(branch="renovate/org.yaml-snakeyaml-1.x")
> DEBUG: branchExists=false(branch="renovate/org.yaml-snakeyaml-1.x")
> DEBUG: dependencyDashboardCheck=undefined(branch="renovate/org.yaml-snakeyaml-1.x")
> DEBUG: recreateClosed is false(branch="renovate/org.yaml-snakeyaml-1.x")
> DEBUG: findPr(renovate/org.yaml-snakeyaml-1.x, Update dependency org.yaml:snakeyaml to v1.33, !open)(branch="renovate/org.yaml-snakeyaml-1.x")
> DEBUG: prAlreadyExisted=false(branch="renovate/org.yaml-snakeyaml-1.x")
> DEBUG: Checking schedule(every 3 months on the first day of the month, Europe/Paris)(branch="renovate/org.yaml-snakeyaml-1.x")
> DEBUG: Found timezone: Europe/Paris(branch="renovate/org.yaml-snakeyaml-1.x")
> DEBUG: Adjusting now for timezone(branch="renovate/org.yaml-snakeyaml-1.x")
> DEBUG: Checking 1 schedule(s)(branch="renovate/org.yaml-snakeyaml-1.x")
> DEBUG: Checking schedule "every 3 months on the first day of the month"(branch="renovate/org.yaml-snakeyaml-1.x")
> 
> DEBUG: Package not scheduled(branch="renovate/org.yaml-snakeyaml-1.x")
> DEBUG: Skipping branch creation as not within schedule(branch="renovate/org.yaml-snakeyaml-1.x")
> 
> DEBUG: remainingBranches=renovate/maven-com.jcraft-jsch-vulnerability,renovate/maven-org.apache.karaf-apache-karaf-vulnerability,renovate/maven-org.quartz-scheduler-quartz-vulnerability,renovate/maven-org.yaml-snakeyaml-vulnerability

**Reply by rarkins** [reply]
Please post the full log for both runs. This branch is related to a vulnerability, so that is maybe related to the problem

**Reply by rknj** [reply]
Here's the full log for when the PR have been autoclosed [renovate_run_969188199_full.log](https://github.com/renovatebot/renovate/files/10432758/renovate_run_969188199_full.log)

**Reply by rknj** [reply]
As the logs history is limitd to 3 days and I don't remember the run number, I cannot get the logs for when they've been reopened

**Reply by rarkins** [reply]
Here is one of the security package rules:

```json
    {
      "matchDatasources": [
        "maven"
      ],
      "matchPackageNames": [
        "com.jcraft:jsch"
      ],
      "matchCurrentVersion": "= 0.1.38",
      "matchFiles": [
        "core/pom.xml"
      ],
      "allowedVersions": "0.1.54",
      "prBodyNotes": [
        "### GitHub Vulnerability Alerts",
        "#### [CVE-2016-5725](https://nvd.nist.gov/vuln/detail/CVE-2016-5725)\n\nDirectory traversal vulnerability in JCraft JSch before 0.1.54 on Windows, when the mode is ChannelSftp.OVERWRITE, allows remote SFTP servers to write to arbitrary files via a ..\\ (dot dot backslash) in a response to a recursive GET command."
      ],
      "isVulnerabilityAlert": true,
      "force": {
        "groupName": null,
        "schedule": "[Circular]",
        "dependencyDashboardApproval": false,
        "stabilityDays": 0,
        "rangeStrategy": "update-lockfile",
        "commitMessageSuffix": "[SECURITY]",
        "branchTopic": "{{{datasource}}}-{{{depName}}}-vulnerability",
        "prCreation": "immediate"
      }
    }
```

```
DEBUG: Looking up com.jcraft:jsch in repository https://repo.maven.apache.org/maven2/
DEBUG: Found 24 new releases for com.jcraft:jsch in repository https://repo.maven.apache.org/maven2/
DEBUG: Looking up com.jcraft:jsch in repository https://devtools.jahia.com/nexus/content/groups/public/
DEBUG: Found 24 new releases for com.jcraft:jsch in repository https://devtools.jahia.com/nexus/content/groups/public/
```

```json
          {
            "datasource": "maven",
            "depName": "com.jcraft:jsch",
            "currentValue": "0.1.38",
            "fileReplacePosition": 73894,
            "registryUrls": [
              "https://repo.maven.apache.org/maven2",
              "https://devtools.jahia.com/nexus/content/groups/public"
            ],
            "depType": "build",
            "depIndex": 86,
            "updates": [
              {
                "bucket": "non-major",
                "newVersion": "0.1.55",
                "newValue": "0.1.55",
                "releaseTimestamp": "2018-11-26T01:15:42.000Z",
                "registryUrl": "https://repo.maven.apache.org/maven2",
                "newMajor": 0,
                "newMinor": 1,
                "updateType": "patch",
                "branchName": "renovate/com.jcraft-jsch-0.x"
              }
            ],
            "warnings": [],
            "versioning": "maven",
            "sourceUrl": "http://git.jcraft.com/jsch.git",
            "homepage": "http://www.jcraft.com/jsch/",
            "currentVersion": "0.1.38",
            "isSingleVersion": true,
            "fixedVersion": "0.1.38"
          }
        ],
        "parent": "pom.xml"
      }
```

Maybe the `= 0.1.38` is not working because of Maven versioning, however it doesn't make sense that it would be flip/flopping between runs. Do you have a log from a run when it reopened the autoclosed remediation PRs?

**Comment by rknj** [comment, marked as answer]
It did not happen again, but I have no idea why as I had to update the config many times. So I guess we can close this discussion.
